### PR TITLE
Content metadata overrides

### PR DESCRIPTION
**Pull request by stonko1994**
## Description
This PR will add the possibility to override `contentMetadata` attributes.

## Tests
New tests about external session managing where added.

### CODE REVIEW

File: BitmovinConvivaAnalytics/Classes/ContentMetadataBuilder.swift [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                contentMetadata.assetName = assetName
+            }
+
+            contentMetadata.viewerId = viewerId
+            contentMetadata.applicationName = applicationName
+
+            if let type = streamType {
+                contentMetadata.streamType = type
+            }
+            if let dur = duration {
```
Comment (jarhoax) [inline BitmovinConvivaAnalytics/Classes/ContentMetadataBuilder.swift]: would it be possible to use `self` here as well? I know we already talked about these namings...

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 2]
```diff
@@ -142,6 +138,32 @@ public final class ConvivaAnalytics: NSObject {
     }
 
     // MARK: - external session handling
+
+    /**
+     Will update the contentMetadata which are tracked with conviva.
+
+     If there is an active session only permitted values will be updated and propagated immediately.
+     If there is no active session the values will set on session creation.
```
Comment (jarhoax) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]:
```suggestion
     If there is no active session the values will be set on session creation.
```

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 3]
```diff
@@ -150,23 +172,18 @@ public final class ConvivaAnalytics: NSObject {
      relies on the players source we can't ensure that all metadata attributes are present at session creation.
      Therefore it could be that there will be a 'ContentMetadata created late' issue after conviva validation.
 
-     - Parameters:
-        - assetName: Will be used as contentMetadata.assetName if no source was loaded before. This overrides the
-                     source assetName. If no source was loaded and no assetName is present this method will throw an
-                     error.
+     If no source was loaded and no assetName was set via updateContentMetadata this method will throw an error.
```
Comment (jarhoax) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]: There could be a source loaded without title, basically.

File: Example/Tests/ExternallyManagedSessionTests.swift [hunk 4]
```diff
@@ -80,7 +83,11 @@ class ExternallyManagedSessionSpec: QuickSpec {
                     }
 
                     it("uses asset anem attribute") {
```
Comment (jarhoax) [inline Example/Tests/ExternallyManagedSessionTests.swift]: typo `anem`

File: README.md [hunk 5]
```diff
@@ -35,34 +35,21 @@ do {
 
 Details about usage of `BitmovinPlayer` can be found [here](https://github.com/bitmovin/bitmovin-player-ios-sdk-cocoapod).
 
-### Advanced Setup
+### Content Metadata handling
 
-You can add more information to your conviva integration via the `ConvivaConfig`:
+If you want to override some content metadata attributes you can do so by adding following:
```
Comment (jarhoax) [inline README.md]:
```suggestion
If you want to override some content metadata attributes you can do so by adding the following:
```

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift (hunk 3 above)
Comment (stonko1994) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]: Uh thanks for that. Good catch 🤔 I guess that can happen on all platforms? So this need to be reworked.

File: BitmovinConvivaAnalytics/Classes/ContentMetadataBuilder.swift (hunk 1 above)
Comment (stonko1994) [reply BitmovinConvivaAnalytics/Classes/ContentMetadataBuilder.swift]: Should be possible

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift (hunk 3 above)
Comment (stonko1994) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]: Actually it's handled. Just the wording is misleading.

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 6]
```diff
@@ -182,7 +183,10 @@ public final class ConvivaAnalytics: NSObject {
 
         if player.config.sourceItem?.itemTitle == nil && contentMetadataBuilder.assetName == nil {
             throw ConvivaAnalyticsError(
-                "AssetName is missing. Load player source first or set assetName via updateContentMetadata"
+"""
```
Comment (jarhoax) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]: Is this multi-line string necessary? and not causing any issues at all?

Comment (stonko1994) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]: Swift lint will complain about the line length. I don't like the multiline string here either. Hm haven't checked the log output format.

Comment (stonko1994) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]: Output is multiline 👎

**Comment by stonko1994** [comment]
@jarhoax please have another review.

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 7]
```diff
@@ -150,23 +172,20 @@ public final class ConvivaAnalytics: NSObject {
...
-        if player.config.sourceItem?.itemTitle == nil && assetName == nil {
-            throw ConvivaAnalyticsError("AssetName is missing. Provide assetName attribute or load player source first")
-        }
-
-        if assetName != nil {
-            metadataOverrides.assetName = assetName
+        if player.config.sourceItem?.itemTitle == nil && contentMetadataBuilder.assetName == nil {
+            throw ConvivaAnalyticsError(
+                // swiftlint:disable:next line_length
+                "AssetName is missing. Load player source (with itemTitle) first. Or set assetName via updateContentMetadata"
```
Comment (jarhoax) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]:
```suggestion
                "AssetName is missing. Load player source (with itemTitle) first or set assetName via updateContentMetadata"
```

**Pull request merged**